src/chapters/auth/intro.rst

```diff
@@ -0,0 +1,133 @@
+
+Flow for Simple Authentication
+------------------------------
+
+Using authentication allows a web application to restrict access to certain pages to known users only. The simplest form of authentication treats all users the same. If a user is signed in, then they are allowed to view *all* restricted pages. Otherwise, they may not view any. With simple authentication, it is not possible to restrict a page to some users while allowing access by others.
+
+Simple authentication requires that for *every* request, the application answers two questions before the request is handled by a controller:
+
+- Is the user asking for a restricted page? 
+- If so, do we know who they are? 
+
+The answers to these questions determine whether or not the user is allowed to view the page.
+
+The logical flow of simple authentication looks like this:
+
+.. figure:: figures/authenticationflowdiagram.png
+   :alt: A diagram showing the the flow of a username and password through the authentication process.
+
+   General authentication flow
+
+Sessions and Cookies
+^^^^^^^^^^^^^^^^^^^^
+.. index:: ! session, ! cookie
+
+The second question above---"Do we know who they are?"---sounds simple enough. However, to determine *which* user is making *which* request turns out to be more complicated. To do this, simple authentication relies on two standard web tools, **sessions** and **cookies**.
+
+A **session** is a collection of temporary data stored *on the server* that corresponds to a specific user. Sessions are often used to store data that an app wants to keep track of in the short term, but not permanently.
+
+.. admonition:: Examples
+
+   Some types of data that might be stored in a session:
+
+   - Items in a shopping cart
+   - Whether or not a user is logged in
+   - Data from a multi-page form, such as a checkout process
+
+A **cookie** is a piece of temporary data stored *in the user's browser*. Cookies are also used to keep short-term data. While a session can hold lots of data entries, a cookie contains only a single key/value pair. However, a browser may hold a large number of cookies.
+
+.. admonition:: Note
+
+   A detailed discussion of cookies is beyond the scope of this book. For more info, read `"HTTP Cookies" on MDN <https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies>`_.
+
+Cookies and sessions work together to enable simple authentication in the following way:
+
+.. index:: 
+   single: cookie; session
+   single: session; ID
+
+#. When a user is authenticated---that is, they provide a valid username/password pair---a session containing their user ID is created on the server. The session has a unique **session ID**, such as ``81LfWG9``.
+#. In order to associate the user with their session, the server returns a **session cookie** to the browser within an HTTP response. The session contains the session ID, and looks something like this: ``session_id: 81LfWG9``. 
```

Comment:
I think this step should be clarified. "..., the server creates a **session cookie** and sends the cookie to the browser...". Also, it should say "The session *cookie* contains the session ID..."